tensorflow/python/ops/map_fn.py

```diff
@@ -375,6 +375,11 @@ def map_fn(fn,
 
   # Flatten the input tensors, and get the TypeSpec for each one.
   elems_flat = nest.flatten(elems)
+
+  # Check in case this is an empty list
+  if len(elems_flat) == 0:
+    raise ValueError("elems must not be empty")
```

Comment:
I looks like the common confusion here is that people expect that when map_fn is applied to a list, that it will map over that list.  To help steer them in the right direction, let's make this message a little more verbose.  How about something along the lines of:

```
raise ValueError(
    "elems must be a Tensor or (possibly nested) sequence of tensors. "
    "Got {}, which does not contain any Tensors.".format(elems))
```